lib/ae_mdw_web/controllers/tx_controller.ex

```diff
@@ -58,7 +61,32 @@ defmodule AeMdwWeb.TxController do
     end
   end
 
-  def read_tx_hash(tx_hash) do
+  #
+  # Private functions
+  #
+  defp add_spendtx_details(response_data, %{"account" => account}) do
+    Enum.map(response_data, fn block ->
+      spend_tx_recipient =
+        if block["tx"]["type"] == @type_spend_tx, do: block["tx"]["recipient_id"]
+
+      if nil != spend_tx_recipient and String.slice(spend_tx_recipient, 0..2) == "nm_" do
```

Comment:
can't we use the id system from the aeternity node here for getting the name account instead of string manipulation?

Reply:
Good point, thanks

Reply:
DBS.map is widely used. Better continue using it's return as is (with encoded human readable recipient_id) comparing with `nm_`.